aea/connections/p2p/stub/connection.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the p2p stub connection."""
+
+import logging
+import os
+import tempfile
+from pathlib import Path
+from typing import Union
+
+from aea.configurations.base import ConnectionConfig, PublicId
+from aea.connections.base import Connection
+from aea.connections.stub.connection import (
+    StubConnection,
```

Comment:
@lrahmani at the moment the framework does not support having one connection depend on another. In this particular case it will work though as the stub connection is distributed in the framework